lib/core/src/send_swap.rs

```diff
@@ -223,18 +192,31 @@ impl SendSwapHandler {
                 create_response.expected_amount,
             )
             .await?;
+        let lockup_tx_id = lockup_tx.txid().to_string();
+
+        self.persister
```

Comment:
Why not persist it only after successful broadcast below (in the `Ok` branch)?

Then I think you wouldn't need to unset it in case of error.

Reply:
We are "locking" the transaction so that the double lockup cannot occur. We persist before by "setting" so that only the current thread can lockup for the same swap. If the broadcast fails, we unset so other threads can try locking up instead.